ctx.go

```diff
@@ -220,6 +223,67 @@ func (c netHttpContext[B]) PathParam(name string) string {
 	return c.Req.PathValue(name)
 }
 
+type PathParamNotFoundError struct {
+	ParamName string
+}
+
+func (e PathParamNotFoundError) Error() string {
+	return fmt.Errorf("param %s not found", e.ParamName).Error()
+}
+
+type PathParamInvalidTypeError struct {
+	Err          error
+	ParamName    string
+	ParamValue   string
+	ExpectedType string
+}
+
+func (e PathParamInvalidTypeError) Error() string {
+	return fmt.Errorf("param %s=%s is not of type %s: %w", e.ParamName, e.ParamValue, e.ExpectedType, e.Err).Error()
+}
+
+type ContextWithPathParam interface {
+	PathParam(name string) string
+}
+
+func PathParamIntErr(c ContextWithPathParam, name string) (int, error) {
+	param := c.PathParam(name)
+	if param == "" {
+		return 0, PathParamNotFoundError{ParamName: name}
+	}
+
+	i, err := strconv.Atoi(param)
+	if err != nil {
+		return 0, PathParamInvalidTypeError{
+			ParamName:    name,
+			ParamValue:   param,
+			ExpectedType: "int",
+			Err:          err,
+		}
+	}
+
+	return i, nil
+}
+
+func (c netHttpContext[B]) PathParamIntErr(name string) (int, error) {
+    return PathParamIntErr(c, name)
+}
+
+func PathParamInt(c ContextWithPathParam, name string) (int) {
```

Comment:
So I don't think you really need this function. You already return 0 with `IntErr`. You don't really need to check the error at all. Just return the param. 

Nitpick none the less.

Reply:
Mmmh that's what we already do with `QueryParamInt` and `QueryParamIntErr`. Maybe a design issue from the start because you're right.

Reply:
Yeah I see that now. Should we create an issue to remove it? I know it’s breaking but we are still pre v1

Reply:
> Maybe a design issue from the start because you're right.

Yeah, before we passed the default value through this function, but now we do by declaring it in the route registration.

I guess we'll do what you suggested before v1.0 yes! Thanks

Reply:
We'll accept this PR as is and create this ticket, if you agree!